pep-9999.rst

```diff
@@ -0,0 +1,257 @@
+
+
+Package index priority
+''''''''''''''''''''''
+
+Tools should respect priority by acting as if all lower priority repositories were not specified when a higher priority repository satisfies the task. Tools that choose not to respect priority MUST display a warning to users when a non-zero priority value is being ignored. It is acceptable for tools to be less efficient in order to handle prioritized repositories, for example, by not performing queries in parallel.
+
+
+Configuration file discovery
+----------------------------
+
+The locations of the configuration files (``config.toml`` and ``auth.toml``) vary slightly across platforms. You may have global, per-user and per-environment configuration:
+
+If multiple configuration files are found, then they are merged in the following order:
+
+1. The global files are read
+2. The per-user files are read
+3. The per-environment files are read
+
+
+Global
+''''''
+
+* Unix: ``/etc/.config/python/``
+* macOS: ``/Library/Application Support/python/``
+* Windows: ``%PROGRAMDATA%\Python``
+
+
+Per-user
+''''''''
+
+* Unix: ``$HOME/.config/python/``
+* macOS: ``$HOME/.config/python/``
+* Windows: ``%APPDATA%\Python\`` (note, this is Roaming app data, not Local)
+
+
+Per-environment
+'''''''''''''''
+
+* Unix: ``sys.prefix``
+* macOS: ``sys.prefix``
+* Windows: ``sys.prefix``
+
+
+Configuration file merging strategy
+'''''''''''''''''''''''''''''''''''
+
+The way the configuration files are merged is by "deep dictionary update", where the ``config.toml`` is updated with any values of the ``auth.toml`` in the particular location. In short, the keys and values are accumulated, where ``auth.toml`` takes precedence in the case of duplicate keys for the given location.
+
+There is no way to _remove_ data from the configuration which has once been defined. However, repositories can be disabled using the ``enable`` attribute.
+
+To do: Python code snippet here to illustrate the exact logic.
+
+
+Backwards Compatibility
+=======================
+
+[Describe potential impact and severity on pre-existing code.]
+
+
+Security Implications
+=====================
+
+[How could a malicious user take advantage of this new feature?]
+
+
+How to Teach This
+=================
+
+[How to teach users, new and experienced, how to apply the PEP to their work.]
+
+
+Reference Implementation
+========================
+
+[Link to any existing implementation and details about its state, e.g. proof-of-concept.]
+
+
+Rejected Ideas
+==============
+
+Base64 encoding for passwords
+'''''''''''''''''''''''''''''
+
+Base64 encoding for passwords were considered, to avoid "over the shoulder" peeking.
+However, this complicates being able to tell the difference between a password and a token.
```

Comment:
I do not think that makes a difference since we can encode a b64 encoded string again. :)  Maybe we should look at maven config for priorart?

Reply:
Yes, but you can't encode on the fly without a computer, I hope? 😉 
The only use case I had in mind here is really when coding on the commute and similar...

What do you think about this?

Later, I would be interested to hear what everyone thinks on this one, especially when sending this draft to the Discourse thread. I _would_ like to see some sort of protection here, but then the token issue, originally raised by @zooba, would have to be solved.

Reply:
My personal thoughts here is that we should not even store this in a plain text file even when base64 encoded. My preference would be to use something like [fernet](https://cryptography.io/en/latest/fernet/) for users that do not use a keyring. That way the key is requested during decrypting. Painful, but cleaner. 

If we do want to use plain text files and not use fernet, then base64 is probably better. We do not need to make any dynamic distinction between token and password really. I am unclear on why the token issue is an issue.